test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetListCommandTest.cs

```diff
@@ -101,6 +101,132 @@ public void ListCommand_WithUserSpecifiedSource()
             }
         }
 
+        // Skip: https://github.com/NuGet/Home/issues/11576
+        [PlatformTheory(Platform.Windows, Platform.Linux, SkipMono = true)]
+        [InlineData(false, false)]
+        [InlineData(false, true)]
+        [InlineData(true, false)]
+        [InlineData(true, true)]
+        public void ListCommand_WithLocalSource_WithMatchText_ReturnLatestVersionOnlyPerPackageId(bool includeMatchingText, bool preRelease)
+        {
+            var nugetexe = Util.GetNuGetExePath();
+
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                // Arrange
+                var packageDirectory = Path.Combine(pathContext.WorkingDirectory, "packageFolder");
+                var solutionFolder = Path.Combine(pathContext.SolutionRoot);
+                string prefix = "newBestPackage";
+                string packageAId = prefix + "A";
+                string packageBId = prefix + "B";
+                var packageFileNameABeta = Util.CreateTestPackage(packageAId, "1.0.0-beta.1.2", packageDirectory);
+                var packageFileNameA1 = Util.CreateTestPackage(packageAId, "1.0.1", packageDirectory);
+                var packageFileNameA2 = Util.CreateTestPackage(packageAId, "2.0.0", packageDirectory);
+                var packageFileNameB1 = Util.CreateTestPackage(packageBId, "1.1.0", packageDirectory);
+                var packageFileNameBPreview = Util.CreateTestPackage(packageBId, "9.2.0-Preview1", packageDirectory);
+
+                // Act
+                var args = $"list -source {packageDirectory} " +
+                            (includeMatchingText ? "best" : string.Empty) + // matching text
+                            (preRelease ? " -prerelease " : string.Empty);
+
+                var r1 = CommandRunner.Run(
+                    nugetexe,
+                    solutionFolder,
+                    args,
+                    waitForExit: true);
+
+                // Assert
+                Assert.Equal(0, r1.ExitCode);
+                var expectedOutput = $"{packageAId} 2.0.0" + Environment.NewLine +
+                                     $"{packageBId} " + (preRelease ? "9.2.0-Preview1" : "1.1.0") + Environment.NewLine;
+                r1.Output.Should().Be(expectedOutput);
+            }
+        }
+
+        [Theory]
+        [InlineData(false, false)]
+        [InlineData(false, true)]
+        [InlineData(true, false)]
+        [InlineData(true, true)]
+        public void ListCommand_WithLocalSource_WithNoMatchingText_ReturnEmpty(bool preRelease, bool allVersions)
```

Comment:
This test also seems like ti would be duplicated. 
It largely matches the some of the test behavior above. 

Note that there's a precedence in the argument order.

Reply:
This one not duplicated with above. This one intentionally searches for `randomId`.
I removed this slow test.